plugins/debuginfo-install.py

```diff
@@ -74,59 +76,170 @@ def configure(self):
                                                            pkg_spec=' '.join(errors_spec))
 
-    def _dbg_available(self, dbgname, package, match_evra):
-        if match_evra:
-            return self.packages_available.filter(
-                name=dbgname,
-                epoch=int(package.epoch),
-                version=str(package.version),
-                release=str(package.release),
-                arch=str(package.arch))
-        else:
-            return self.packages_available.filter(
-                name=dbgname,
-                arch=str(package.arch))
-
-    def _di_install(self, package):
-        for dbgname in [package.debug_name, package.source_debug_name]:
-            if dbgname in self.dbgdone:
-                break
-            if self._dbg_available(dbgname, package, True):
-                di = "{0}-{1}:{2}-{3}.{4}".format(
-                    dbgname,
-                    package.epoch,
-                    package.version,
-                    package.release,
-                    package.arch)
-                self.base.install(di)
-            elif self._dbg_available(dbgname, package, False):
-                di = "{0}.{1}".format(dbgname, package.arch)
-                self.base.install(di)
-            else:
-                continue
-            self.dbgdone.append(dbgname)
-            break
-        else:
-            logger.info(
-                _("Could not find debuginfo for package: %s") % package)
+    def _install_debug_from_system(self, debug_name, package):
+        query = self.base.sack.query().filter(name=debug_name,
+                                              epoch=package.epoch,
+                                              version=package.version,
+                                              release=package.release,
+                                              arch=package.arch)
+
+        if query:
+            self._install(query)
+            return True
+
+        return False
+
+    def _install_debug(self, debug_name, base_nevra):
+        kwargs = {}
+
+        # if some part of EVRA was specified in the argument, add it as a filter
+        if base_nevra.epoch is not None:
+            kwargs["epoch__glob"] = base_nevra.epoch
+        if base_nevra.version is not None:
+            kwargs["version__glob"] = base_nevra.version
+        if base_nevra.release is not None:
+            kwargs["release__glob"] = base_nevra.release
+        if base_nevra.arch is not None:
+            kwargs["arch__glob"] = base_nevra.arch
+
+        query = self.base.sack.query().filter(name=debug_name, **kwargs)
+
+        if query:
+            self._install(query)
+            return True
+
+        return False
+
+    def _install_debug_no_nevra(self, debug_name, packages):
+        query = self.base.sack.query().filterm(
+            nevra_strict=["{}-{}.{}".format(debug_name, p.evr, p.arch) for p in packages])
```

Comment:
Please could you rename p to pkg. This is more common name in the project

Reply:
I would actually rather prefer a single-letter variable here. The scope is limited to this single line, so there is no doubt what it represents and the single letter makes the line shorter and easier to read. I think single letters are common in constructs like list comprehension and good to use there...

Reply:
I could, though, rename:
`packages -> pkgs`
`package -> pkg`

Throughout the code. I does save quite a few characters overall and if it's sort of a convention...

Reply:
Proposed change `packages -> pkgs` `package -> pkg` is fine.